apps/pipe-intermediate/src/app/app.component.ts

```diff
@@ -1,14 +1,41 @@
 import { NgFor } from '@angular/common';
 import { Component } from '@angular/core';
+import { Pipe, PipeTransform } from '@angular/core';
+
+@Pipe({
+  name: 'compute',
+  pure: true,
+  standalone: true,
+})
+export class ComputePipe implements PipeTransform {
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  transform(value: any, funcName: 'showName' | 'isAllowed'): string {
+    return funcName === 'isAllowed'
+      ? this.isAllowed(value.age, value.isFirst)
+      : this.showName(value.name, value.index);
+  }
+
+  private showName(name: string, index: number) {
+    return `${name} - ${index}`;
+  }
+
+  private isAllowed(age: number, isFirst: boolean) {
+    if (isFirst) {
+      return 'always allowed';
+    } else {
+      return age > 25 ? 'allowed' : 'declined';
+    }
+  }
+}
```

Comment:
I think my readme is still unclear. 

All functions need to stay inside the component. The pipe should only be a wrapper (a very generic pipe that you can use everywhere in any component without having to add a new function and condition to it)

Reply:
I couldn't get what are you saying and yeah readme is always very clear, can you show me some examples?

Reply:
This exercice is to create a pipe to wrap a function to use the memoization property of a pure pipe. 

If you want to use this pipe on another component with different function, you will need to add those function inside the pipe and that's not what we want. 

You could only pass the callback of the function as params to your pipe

like 

```typescript
transform(value: any, func: Function) {
return func(value)
}
```